nvflare/private/fed/server/training_cmds.py

```diff
@@ -119,10 +120,16 @@ def _shutdown_app_on_clients(self, conn: Connection) -> bool:
         replies = self.send_request_to_clients(conn, message)
         self.process_replies_to_table(conn, replies)
 
-        err = engine.remove_clients(clients)
-        if err:
-            conn.append_error(err)
-            return False
+        clients_to_be_removed = set(clients)
+        for r in replies:
+            if r.reply.get_header(MsgHeader.RETURN_CODE) == ReturnCode.ERROR:
+                clients_to_be_removed.remove(r.client_token)
+
+        if clients_to_be_removed:
+            err = engine.remove_clients(clients_to_be_removed)
+            if err:
+                conn.append_error(err)
+                return False
```

Comment:
This logic is not correct. when clients_to_be_removed != clients, it means some clients responded error. It needs to return False. 
In the extreme case, all the clients have error return code, the clients_to_be_removed will be empty. With the current logic, it will return True.

Reply:
yes, thanks for the review, I just update this PR.